pkg/graveler/ref/manager.go

```diff
@@ -47,23 +47,75 @@ func NewKVRefManager(executor batch.Batcher, kvStore kv.StoreMessage, db db.Data
 }
 
 func (m *KVManager) GetRepository(ctx context.Context, repositoryID graveler.RepositoryID) (*graveler.Repository, error) {
-	return m.db.GetRepository(ctx, repositoryID)
+	repo := graveler.RepositoryData{}
+	_, err := m.kvStore.GetMsg(ctx, graveler.RepoPartition(), []byte(graveler.RepoPath(repositoryID)), &repo)
+	if err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			err = graveler.ErrRepositoryNotFound
+		}
+		return nil, err
+	}
+	return graveler.RepoFromProto(&repo).Repository, nil
+}
+
+func (m *KVManager) createBareRepository(ctx context.Context, repositoryID graveler.RepositoryID, repository graveler.Repository) error {
+	repoRecord := &graveler.RepositoryRecord{
+		RepositoryID: repositoryID,
+		Repository:   &repository,
+	}
+	repo := graveler.ProtoFromRepo(repoRecord)
+	err := m.kvStore.SetMsgIf(ctx, graveler.RepoPartition(), []byte(graveler.RepoPath(repositoryID)), repo, nil)
+	if err != nil {
+		if errors.Is(err, kv.ErrPredicateFailed) {
+			err = graveler.ErrNotUnique
+		}
+		return err
+	}
+	return nil
 }
 
 func (m *KVManager) CreateRepository(ctx context.Context, repositoryID graveler.RepositoryID, repository graveler.Repository, token graveler.StagingToken) error {
-	return m.db.CreateRepository(ctx, repositoryID, repository, token)
+	firstCommit := graveler.NewCommit()
+	firstCommit.Message = graveler.FirstCommitMsg
+	firstCommit.Generation = 1
+	// commitID := m.addressProvider.ContentAddress(firstCommit)
+
+	err := m.createBareRepository(ctx, repositoryID, repository)
+	if err != nil {
+		return err
+	}
+
+	commitID, err := m.AddCommit(ctx, repositoryID, firstCommit)
+	if err != nil {
+		return err
+	}
+
+	err = m.CreateBranch(ctx, repositoryID, repository.DefaultBranchID, graveler.Branch{CommitID: commitID, StagingToken: token})
```

Comment:
so we decided that if we create a repository, succeed in creating the repository and fail in the commit or branch and then try to create another repository with the same name it will fail?

Reply:
Absolutely not!
I still need to figure out the correct (or, should I say, best) error handling in that case.
Please refer to my proposition on regarding the `repositroy_status`